core/src/mindustry/world/blocks/defense/Door.java

```diff
@@ -126,7 +126,7 @@ public void draw(){
 
         @Override
         public Cursor getCursor(){
-            return SystemCursor.hand;
+            return player.team() == team ? SystemCursor.hand : SystemCursor.arrow;
```

Comment:
it should probably use one of the built-in interaction checks:

https://github.com/Anuken/Mindustry/search?q=canInteract&type=

```suggestion
            return interactable(player.team()) ? SystemCursor.hand : SystemCursor.arrow;
```